pip/__init__.py

```diff
@@ -181,19 +181,51 @@ def parseopts(args):
     # the subcommand name
     cmd_name = args_else[0]
 
-    if cmd_name not in commands_dict:
-        guess = get_similar_commands(cmd_name)
-
-        msg = ['unknown command "%s"' % cmd_name]
-        if guess:
-            msg.append('maybe you meant "%s"' % guess)
-
-        raise CommandError(' - '.join(msg))
-
     # all the args without the subcommand
     cmd_args = args[:]
     cmd_args.remove(cmd_name)
 
+    # Autocorrect command name
+    if cmd_name not in commands_dict:
+        # MARK: The following should be loaded from the configuration file
+        #       in the future. For now, it can stay like this, I guess.
+        wait_time = 2          # Numeric, only first post decimal significant
+        suggest_cut_off = 0.6  # float between 0-1
+        replace_cut_off = 0.8  # float between 0-1
+
+        assert suggest_cut_off <= replace_cut_off, \
+            "autocorrect - suggestions cut off value invalid!"
+
+        score, guess = get_similar_command(cmd_name)
+
+        if guess is None:  # nothing similar
+            err_msg = 'pip does not have a command "%s"' % cmd_name
+        elif score < replace_cut_off:  # not similar enough
+            err_msg = (
+                'pip does not have a command "%s" - did you mean "%s"?'
+            ) % (cmd_name, guess)
+        else:  # close enough - no error
+            err_msg = None
+
+        if err_msg is not None:
+            raise CommandError(err_msg)
+
+        msg = (
+            'You called a pip command named "%s" which does not exist.\n'
+            'Assuming you meant "%s", pip will continue in %.1f seconds...'
+        )
+
+        logger.warn(msg, cmd_name, guess, wait_time)
+        try:
+            # time.sleep in a loop because KeyboardInterrupt is raised after
+            # it returns
+            for i in range(int(wait_time * 10)):
+                time.sleep(0.1)
+        except KeyboardInterrupt:
```

Comment:
Hmm, you are right. Then, it would be nice to print the same error messages: https://github.com/pypa/pip/blob/bbe99ce/pip/basecommand.py#L236-L237 and `sys.exit(pip.status_codes.ERROR)`, to be consistent.